app/src/main/java/com/example/sharingang/items/CachedItemRepository.kt

```diff
@@ -0,0 +1,51 @@
+package com.example.sharingang.items
+
+import androidx.lifecycle.LiveData
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+
+class CachedItemRepository @Inject constructor(
+    private val itemDao: ItemDao,
+    private val store: ItemStore
+) : ItemRepository {
+
+    override fun items(): LiveData<List<Item>> {
+        return itemDao.getAllItems()
+    }
+
+    override suspend fun refreshItems() {
+        // This is necessary, since you want to avoid doing this work on the main thread
+        withContext(Dispatchers.IO) {
+            val newItems = store.getAllItems()
+            itemDao.insert(newItems)
```

Comment:
Ideally we shouldn't call getAllItems all the time.
Maybe on addItem, just get the the added Item from the store and similarly for the other actions.

Reply:
On the other hand, this provides a good occasion to refresh the items automatically, and you need network connection to do these things *anyways*.

I think if we're worried about pulling all the items being too expensive, then we'd need to paginate requests *anyways* at that point, since the initial refresh of the items would also be too slow.

Reply:
One concern would be latency, but this is hidden because you can go on with your business while you've sent off the action, and thus the refresh, to firebase

Reply:
You're right and maybe when we'll have search, we will just retrieve a few items and update that